loader/loader.go

```diff
@@ -64,13 +60,9 @@ func (l *Loader) onRuntimeChanged() {
 	l.nextSnapshot = snapshot.New()
```

Comment:
How is it safe that this happens outside of the mutex?

Reply:
It's safe since `onRuntimeChanged()` and  is only called here and its the only thing that modifies `nextSnapshot`:

https://github.com/lyft/goruntime/blob/e963ba27ca2fdd94d442cb87af99c5582e70da8b/loader/loader.go#L195-L204

Reply:
I do not find the argument that it is safe because we currently use it in a safe way very compelling. If we started calling `onRuntimeChanged` concurrently, there would be a race. Since `walkDirectoryCallback` seems to be the only other place we're using this var, how about making it a local to `onRuntimeChanged`, and passing a pointer into `wDC`?

Reply:
Btw, this thread should not hold up the PR if you don't care about this, just providing some unsolicited input :)

Reply:
>  If we started calling onRuntimeChanged concurrently, there would be a race

This was also true when we used the mutex. We can't change the function signature to `walkDirectoryCallback` since it has to match [filepath.WalkFunc](https://golang.org/pkg/path/filepath/#WalkFunc) to be used by `filepath.Walk()`. Fortunately, the `Loader` only watches one directory so it will likely never need to work in parallel (we also change runtime by swapping out symlink to the top-level `data` dir so once one thing changes so we'd never need a feature like walking different sub dirs in parallel).

That said, this library always makes me do a double take every time I work in it since it skirts on the edge of safety and in some cases allows for unsafe things ([snapshot.go#L109-L115](https://github.com/lyft/goruntime/blob/2c2ecd265bb8da97bc5f3d05536429213a6459eb/snapshot/snapshot.go#L109-L115)).